src/meta_schedule/feature_extractor/per_block_feature.cc

```diff
@@ -314,25 +311,109 @@ class PerBlockFeatureCollector : private StmtVisitor {
         cache_line_bytes_(cache_line_bytes),
         arith_intensity_curve_num_samples_(arith_intensity_curve_num_samples) {}
 
+  void VisitStmt_(const BlockRealizeNode* realize) final {
+    if (!scopes_.empty()) {
+      ordered_blocks_.push_back(realize);
+    }
+    scopes_.push_back(realize);
+    dfs_path_.push_back(realize);
+    StmtVisitor::VisitStmt_(realize);
+    dfs_path_.pop_back();
+    scopes_.pop_back();
+    if (scopes_.empty()) {
+      return;
+    }
+    // Get the ancestor loops from inner to outer, up to the parent scope
+    ForVec loops;
+    for (auto iter = dfs_path_.rbegin(); iter != dfs_path_.rend(); ++iter) {
+      const StmtNode* stmt = *iter;
+      if (stmt->IsInstance<ForNode>()) {
+        loops.push_back(static_cast<const ForNode*>(stmt));
+      }
+    }
+    Feature& feature = block_features_[realize];
+    feature.block_realize = realize;
+    feature.group1 = std::make_unique<group1::Feature>(realize, loop_nest_, is_gpu_);
+    if (!scopes_.empty()) {
+      AddArithOpsToScope(&feature.group1->arith_ops);
+    }
+    // TODO: add groups 2-5
+  }
+
   void VisitStmt_(const ForNode* loop) final {
     int64_t auto_unroll;
+    const int64_t* extent = GetLoopIntExtent(loop);
     ForVec* for_vec = loop_nest_.Push(loop, &auto_unroll);
+    if ((extent && (*extent != 1)) || for_vec) {
+      outer_loop_prod_ *= *extent;
+      dfs_path_.push_back(loop);
+      analyzer_.Bind(loop->loop_var, loop->min);
```

Comment:
It's uncertain for me if we really want to do value binding here.

Some background:
- The analyzer is a tool that does integer set analysis, which provides basic functionality like proving whether or not two expressions are equal
- `Bind` means letting the analyzer know that a variable's value is always a constant - in our particular case, enforcing the assumption that `loop->loop_var == loop->min`

However, when the loop extent is not 1, for example,
```python
for i in range(0, 10): ... # extent is 10
```
binding `loop->loop_var` (i.e. `i`) to `loop->min` (i.e. `0`) doesn't seem to be a reasonable assumption.

Therefore, perhaps we should postpone this binding to follow-up PRs once we have better understanding of what this binding is actually used for.

Reply:
okay, I commented it out